components/automate-cli/pkg/diagnostics/integration/iam_projects.go

```diff
@@ -61,8 +61,9 @@ type RulesInfo struct {
 }
 
 type iamProjectSave struct {
-	ID     string `json:"id"`
-	RuleID string `json:"rule_id"`
+	ID      string `json:"id"`
+	RuleID  string `json:"rule_id"`
+	Skipped bool   `json:"skipped"`
```

Comment:
The diagnostics were failing in the following case:
- diagnostics run on v1, skipping iam-projects, iam-roles, and iam-policies-v2 since those are v2-only
- force-upgrade to IAM v2
- diagnostics run again, this time not skipping projects/roles/policies-v2. but no data was generated while on v1, so the diagnostics fail to verify any data.

This is only a problem when we break up the diagnostics with the options `--skip-cleanup` `--skip-generate`, which only happens during integration tests. A customer who just runs `chef-automate diagnostics run` will never hit this edge case 

To solve this problem, we took a cue from @danielsdeleo's solution in the applications diagnostic - we keep track of whether or not the diagnostic was skipped by adding a bool to the `save` state. If it was skipped, we can return early from the Verify and Cleanup steps instead of failing.